youtube_dl/extractor/tvnoe.py

```diff
@@ -0,0 +1,54 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import json
+import re
+from youtube_dl.compat import compat_urlparse
+from .common import InfoExtractor
+from ..utils import clean_html, get_element_by_class
+
+SRC_PATTERN = r'<script.+jwplayer.*sources:\s*(\[[^]]+\])'
+
+
+class TVNoeIE(InfoExtractor):
+    _VALID_URL = r'https?://(www\.)?tvnoe\.cz/video/(?P<id>[0-9]+)'
+    _TEST = {
+        'url': 'http://www.tvnoe.cz/video/10362',
+        'md5': 'aee983f279aab96ec45ab6e2abb3c2ca',
+        'info_dict': {
+            'id': '10362',
+            'ext': 'mp4',
+            'series': 'Noční univerzita',
+            'title': 'prof. Tomáš Halík, Th.D. - ' +
+                     'Návrat náboženství a střet civilizací',
+            'description': 'md5:f337bae384e1a531a52c55ebc50fff41',
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        iframe_url = self._search_regex(r'<iframe[^>]+src="([^"]+)"',
+                                        webpage, 'iframe src attribute')
+
+        ifs_page = self._download_webpage(iframe_url, video_id)
+        ifs_sources = self._search_regex(SRC_PATTERN, ifs_page, 'sources')
+        ifs_sources = json.loads(re.sub(r'(file|label):', r'"\1":',
+                                        ifs_sources))
+
+        assert len(ifs_sources) > 0
+
+        formats = [{'url': compat_urlparse.urljoin(iframe_url, source['file']),
+                    'format_id': source['label']}
+                   for source in ifs_sources]
```

Comment:
> js_to_json in utils.py should do the work.

Please, could you let me there my **working** code instead of force to run HTML pages through random RegExps and pretend it is more robust? I understand your NIH syndrome and pride over your utils, but I have already wasted couple of hours on making my code (IMO) more ugly. Could we stop, please?

http://paste.fedoraproject.org/419766/16236147/ leads (of course) to
```
matej@mitmanek: youtube-dl (tvnoe *)$ python test/test_download.py TestDownload.test_TVNoe
[TVNoe] 10362: Downloading webpage
[TVNoe] 10362: Downloading webpage
E
======================================================================
ERROR: test_TVNoe (__main__.TestDownload)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/test_download.py", line 141, in test_template
    force_generic_extractor=params.get('force_generic_extractor', False))
  File "/home/matej/archiv/knihovna/repos/youtube-dl/youtube_dl/YoutubeDL.py", line 691, in extract_info
    ie_result = ie.extract(url)
  File "/home/matej/archiv/knihovna/repos/youtube-dl/youtube_dl/extractor/common.py", line 347, in extract
    return self._real_extract(url)
  File "/home/matej/archiv/knihovna/repos/youtube-dl/youtube_dl/extractor/tvnoe.py", line 43, in _real_extract
    js_to_json(ifs_page), video_id, require_title=False)
  File "/home/matej/archiv/knihovna/repos/youtube-dl/youtube_dl/extractor/jwplatform.py", line 29, in _extract_jwplayer_data
    self._find_jwplayer_data(webpage), video_id)
  File "/home/matej/archiv/knihovna/repos/youtube-dl/youtube_dl/extractor/common.py", line 549, in _parse_json
    return json.loads(json_string)
  File "/usr/lib64/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
TypeError: expected string or buffer

----------------------------------------------------------------------
Ran 1 test in 1.052s

FAILED (errors=1)
matej@mitmanek: youtube-dl (tvnoe *)$
```